coordinator/persistence-cassandra-4.1/src/main/java/io/stargate/db/cassandra/impl/StargateSystemKeyspace.java

```diff
@@ -0,0 +1,126 @@
+package io.stargate.db.cassandra.impl;
+
+import com.google.common.collect.ImmutableList;
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLStatement;
+import org.apache.cassandra.cql3.QueryProcessor;
+import org.apache.cassandra.cql3.statements.SelectStatement;
+import org.apache.cassandra.db.SystemKeyspace;
+import org.apache.cassandra.db.virtual.*;
+import org.apache.cassandra.dht.IPartitioner;
+import org.apache.cassandra.dht.Token.TokenFactory;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.schema.*;
+import org.apache.cassandra.transport.ProtocolVersion;
+import org.apache.cassandra.utils.FBUtilities;
+import org.apache.cassandra.utils.MurmurHash;
+
+public class StargateSystemKeyspace extends VirtualKeyspace {
```

Comment:
Odd; neither C-4 nor DSE-6.8 variants extend `VirtualKeyspace` altho that type exists on their native backends.